lib/todo.dart

```diff
@@ -30,4 +33,59 @@ class Todo {
   String toJson() => json.encode(toMap());
 
   factory Todo.fromJson(String source) => Todo.fromMap(json.decode(source));
+
+  Todo copyWith({
+    String? title,
+    String? description,
+    bool? isDone,
+  }) {
+    return Todo(
+      title: title ?? this.title,
+      description: description ?? this.description,
+      isDone: isDone ?? this.isDone,
+    );
+  }
+}
+
+class TodoListNotifier extends ValueNotifier<List<Todo>> {
+  TodoListNotifier(super.state) {
+    _initializeTodos();
+  }
+
+  late SharedPreferences prefs;
+
+  Future<void> _initializeTodos() async {
+    prefs = await SharedPreferences.getInstance();
```

Comment:
glued dependency FYI, maybe fine in this showcase.

Reply:
What does "glued" mean?

Reply:
you can't fake it out for tests as it's now tied to the class

Reply:
I see. Yes it is glued then, but it's done on purpose here for simplicity. I don't think it's worth going into software architecture and abstraction at this point. 

If the course expands into testing as architecture topics, we can walk through "unglueing" it in that section